### ISSUE TITLE
loss nan in Genomic Benchmark

### ISSUE THREAD

**Issue by multydoffer** [kuleshov-group/caduceus#32]
I am using the 470k released weight to finetune on genomic benchmark, but encounter nan loss on both ps and ph models. If the learing rate is set to 1e-3, the nan will occur in the 2nd epoch. What's going on? I've tried other size models, they can run smoothly

**Comment by yair-schiff** [comment]
I would use a lower LR. These models were trained with a cosine decay schedule that ended at `1e-4` after 10k steps, so I would start with that at the high end and try even lower for further fine-tuning

**Comment by multydoffer** [comment]
However, 1e-7 still produce nan. This is wired, cuz I have clip the gradient also. And I notice that in your paper you state that the lr is 1e-3 right?

**Comment by multydoffer** [comment]
And the lr is already set with cosine decay scheduler in the 1e-3 setting

**Comment by multydoffer** [comment]
> I would use a lower LR. These models were trained with a cosine decay schedule that ended at `1e-4` after 10k steps, so I would start with that at the high end and try even lower for further fine-tuning
However, 1e-7 still produce nan. This is wired, cuz I have clip the gradient also. And I notice that in your paper you state that the lr is 1e-3 right?
And the lr is already set with cosine decay scheduler in the 1e-3 setting

**Comment by multydoffer** [comment]
However, 1e-7 still produce nan. This is wired, cuz I have clip the gradient also. And I notice that in your paper you state that the lr is 1e-3 right?
And the lr is already set with cosine decay scheduler in the 1e-3 setting

> I am using the 470k released weight to finetune on genomic benchmark, but encounter nan loss on both ps and ph models. If the learing rate is set to 1e-3, the nan will occur in the 2nd epoch. What's going on? I've tried other size models, they can run smoothly
